holoviews/element/raster.py

```diff
@@ -73,14 +70,29 @@ def __getitem__(self, slices):
         if all(slc_types):
             return self.clone(data, extents=None)
         elif not any(slc_types):
-            return toarray(data, index_value=True)
+            return data
         else:
             return self.clone(np.expand_dims(data, axis=slc_types.index(True)),
                               extents=None)
 
 
-    def _coord2matrix(self, coord):
-        return int(round(coord[1])), int(round(coord[0]))
+    def dimension_values(self, dim, expanded=True, flat=True):
+        """
+        The set of samples available along a particular dimension.
+        """
+        dim_idx = self.get_dimension_index(dim)
+        if not expanded and dim_idx == 0:
+            return np.array(range(self.data.shape[1]))
+        elif not expanded and dim_idx == 1:
+            return np.array(range(self.data.shape[0]))
+        elif dim_idx in [0, 1]:
+            values = np.mgrid[0:self.data.shape[1], 0:self.data.shape[0]][dim_idx]
+            return values.flatten() if flat else values
+        elif dim_idx == 2:
+            arr = self.data.T
+            return arr.flatten() if flat else arr
+        else:
+            return super(Raster, self).dimension_values(dim)
```

Comment:
Might be worth leaving a comment saying the last branch is assuming you are left with cdims...